WIP: Refactor

has2k1 opened the pull request:
This PR does the refactoring necessary for #5. Still to do are:

- [x] Update documentation 
- [x] Add tests

The documentation should have a table for each supported package. The rows in the table should be entries for a supported Model and 3 added rows for each of tidy function to indicate support for that function.

With pytest, there should be a way to define a base test class that has test methods, where the tests subclass the base-class and create result objects that get automatically tested.

has2k1 committed:
Register & move worker functions into submodules

Use singledispatch to invoke them.

has2k1 committed:
Autoimport the subpackages to build the registries

Implementations for Model types can be added in pybroom
subpackages they are automatically registered if the Model
package is installed.

has2k1 commented:
@tritemio, can you enable travis for this repo.

Also, there is commented out code for versioneer. Any reasons why it did not work out?

tritemio commented:
@has2k1, great! It looks good at first glance. I enabled Travis. Regarding versioneer, I don't remember, maybe I didn't want to add it because its size was much larger than the entire pybroom.py. Now, it makes sense to enable it anyway.

has2k1 committed:
Make param_names for scipy.optimize optional

This is a good default but it has been made in anticipation,
so that automated tests -- when implemented --  can be uniform
across all models.

has2k1 commented:
@tritemio, I've made a small API change in the last commit.

About a decision in that commit, are automatic parameter names *p0, p1, p2, ... pn* a surprise to the user? I think I have seen names as such, but I do not recall where.

has2k1 committed:
Add tests

The BaseTest abstract class does all the testing. The subclasses
define input.

has2k1 committed:
Add .travis.yml and a Makefile

has2k1 committed:
Fix linting issue

has2k1 committed:
DOC: Update function names and use autosummary

Having the function documentation on separate pages
keeps the API page tidy :).

has2k1 committed:
Remove lmfit from the requirements

has2k1 committed:
Add versioneer

has2k1 committed:
Remove manual version specification

has2k1 committed:
Add sphinx_rtd_theme to documentation requirements

sphinx does not ship with the rtd theme anymore.

has2k1 committed:
Add pandoc install on travis for the doc. test

has2k1 committed:
Alter .travis.yml to try and get doc test to pass
